src/palace/manager/celery/tasks/boundless.py

```diff
@@ -1,520 +1,112 @@
-        )
-        log_run_end_message()
-
-
-def process_book(
-    task: Task,
-    _db: Session,
-    api: BoundlessApi,
-    metadata: BibliographicData,
-) -> None:
-    edition, new_edition, license_pool, new_license_pool = api.update_book(
-        bibliographic=metadata,
-    )
-
-    task.log.info(
-        f"Edition (id={edition.id}, title={edition.title}) {'created' if new_edition else 'updated'}. "
-        f"License pool (id={license_pool.id}) {'created' if new_license_pool else 'updated'}."
-    )
-
-
-def _redis_lock_list_identifiers_for_import(
-    client: Redis, collection_id: int
-) -> RedisLock:
-    return RedisLock(
-        client,
-        lock_name=[
-            f"ListIdentifiersForImport",
-            Collection.redis_key_from_id(collection_id),
-        ],
-    )
-
-
-def get_collections_by_protocol(
-    task: Task,
-    session: Session,
-    protocol_class: type[BaseCirculationAPI[SettingsType, LibrarySettingsType]],
-) -> list[Collection]:
-    registry = task.services.integration_registry.license_providers()
-    return (
-        session.execute(
-            Collection.select_by_protocol(protocol_class, registry=registry)
+        return BoundlessImporter(
+            session, collection, registry, import_all, identifier_set
+        ).import_collection(
+            apply_bibliographic=apply.bibliographic_apply.delay,
         )
-        .scalars()
-        .all()
-    )
 
 
 @shared_task(queue=QueueNames.default, bind=True)
-def reap_all_collections(
-    task: Task,
-    delay_in_seconds_between_reap_collection_tasks: int = MAX_DELAY_IN_SECONDS_BETWEEN_REAP_TASKS,
-) -> None:
+def reap_all_collections(task: Task, *, import_all: bool = False) -> None:
     """
-    A shared task that kicks off a reap_collection task for each Boundless collection.
-    Palace Managers can be configured with many Boundless collections which may have identical or nearly identical
-    lists of identifiers.  As a result reaping Boundless collections in parallel can lead to frequent
-    deadlocks when launched in parallel.
-
-    The default delay of 20 seconds between launching of reap_collection sub-tasks is based on
-    observations of performance in a local environment.  I observed that reaping in batches 10 seconds
-    can take from 2-4 seconds in my development environment. The production environment was
-    observed to be about 2 to 3 times slower.  Therefore,a 20 second delay between task initiation provides a reasonable
-    empirically based time buffer such that batches of boundless identifiers are not reaped simultaneously and won't likely
-    cause a pile-up in the case of occasional deadlocks.  With the redis visibility_timeout set to one hour, this will
-    allow for a maximum of 180 collections.  Due to the visibility_timeout constraint,  20 seconds will be the maximum
-    delay:  any value over 20 will be ignored and the default will be used in its place.
```

Comment:
I think this will also be handled by the collection level locks, but since its a concurrency thing, its hard to tell until it goes out to a busy server.